lib/jekyll-github-metadata/metadata_drop.rb

```diff
@@ -10,6 +10,11 @@ class MetadataDrop < Jekyll::Drops::Drop
 
       mutable true
 
+      def initialize(obj, original_config = nil)
+        super(obj)
+        @original_config = original_config
```

Comment:
Hello @dunkmann00.
What version of this plugin are you using? To my knowledge, the issue with Jekyll Cache calling `config.inspect` was fixed once the plugin stopped depending on `site.config["github"]` from v2.15.0 via https://github.com/jekyll/github-metadata/commit/a771d0d7457a7bff42b129ce0bed70c74514a506.
Therefore, only the Drop being cast into a Hash should trigger the resolution of all drop keys.

*Just asking for clarity.*

Reply:
Hello @ashmaroli! I am using `2.15.0`. The change via the commit you reference fixes the problem on the initial load and render of the site. However, when the site is rebuilt automatically while running `jekyll serve`, the problem presents itself again.

The original fix waits to inject `site.config["github"]` until after `process` and therefore `reset` is called. But when `jekyll-watch` notices a change and calls `process`, `site.config["github"]` has already been injected. Since it is already injected, we see the original problem occurring once again.

Reply:
```ruby
    Jekyll::Hooks.register :site, :pre_render do |_site, payload|
      SiteGitHubMunger.global_munger.inject_metadata!(payload)
    end
```

Is there a post_render hook where we could un-inject this?

Reply:
🥳 That seems like a great idea! I think at one point this thought crossed my mind but I didn't understand enough of how the library worked so I didn't pursue it. But looking again this totally seems doable. I'll update the PR with this change.